django-stubs/db/backends/base/base.pyi

```diff
@@ -39,24 +43,23 @@ class BaseDatabaseWrapper:
     savepoint_ids: Any = ...
     commit_on_exit: bool = ...
     needs_rollback: bool = ...
-    close_at: Optional[Any] = ...
+    close_at: Optional[float] = ...
     closed_in_transaction: bool = ...
     errors_occurred: bool = ...
     allow_thread_sharing: bool = ...
-    run_on_commit: List[Any] = ...
+    run_on_commit: List[Callable[[], None]] = ...
     run_commit_hooks_on_set_autocommit_on: bool = ...
-    execute_wrappers: List[Any] = ...
+    execute_wrappers: List[Callable[[Any, Any, Any, Any, Any], None]] = ...
```

Comment:
Why is it `Callable[[Any, Any, Any, Any, Any], None]`? Can we be more specific?

Reply:
This is one of those rabbit holes... The documentation is hand-wavy https://docs.djangoproject.com/en/3.2/topics/db/instrumentation/ and I still don't understand what the return type is supposed to be, 
```python
_SQLType = Union[None, bool, int, float, Decimal, str, bytes, datetime, UUID]
_ExecuteParameters = Optional[Union[Sequence[_SQLType], Mapping[str, _SQLType]]]
_ExecuteWrapper = Callable[
    [
        Callable[[str, Union[_ExecuteParameters, Sequence[_ExecuteParameters]], bool, Dict[str, Any]], Any],
        str,
        Union[_ExecuteParameters, Sequence[_ExecuteParameters]],
        bool,
        Dict[str, Any],
    ],
    Any,
]
```

And this is probably still wrong because I believe `_SQLType` could also contain DB-backend-specific types like `cx_Oracle.Binary`.

Reply:
I guess hinting the type of the `params` argument is counter-productive. If users want to use this, unless they add an `if TYPE_CHECKING` guard and import the internal `_ExecuteParameters` type, they would need to annotate that argument as the type soup:
`Union[Optional[Union[Sequence[Union[None, bool, int, float, Decimal, str, bytes, datetime, UUID]], Mapping[str, Union[None, bool, int, float, Decimal, str, bytes, datetime, UUID]]]], Sequence[Optional[Union[Sequence[Union[None, bool, int, float, Decimal, str, bytes, datetime, UUID]], Mapping[str, Union[None, bool, int, float, Decimal, str, bytes, datetime, UUID]]]]]]`

And it would break every time something gets added to `_SQLType`.

I'll just use `Any` for now.